File: management-domain/Digital-marketing/Backend/Nodes/marketing-automation-service/social-automation-service/src/interfaces/rest/controllers/health.controller.ts
import { Controller, Get } from '@nestjs/common';

@Controller('health')
export class HealthController {
  @Get() health() { return { status: 'ok', service: 'social-automation-service', timestamp: new Date().toISOString() }; }
  @Get('ready') ready() { return { status: 'ready' }; }
  @Get('live') live() { return { status: 'alive' }; }
}

File: foundation-domain/centralized-configuration/config-server/src/main/java/com/gogidix/centralconfiguration/configserver/interfaces/rest/HealthController.java
package com.gogidix.centralconfiguration.configserver.interfaces.rest;

import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.time.LocalDateTime;
import java.util.HashMap;
import java.util.Map;

/**
 * Health check controller for config server.
 */
@RestController
@RequestMapping("/actuator")
@Tag(name = "Health", description = "Health check endpoints")
public class HealthController {

    @GetMapping("/health")
    @Operation(summary = "Health check", description = "Returns the health status of the service")
    public ResponseEntity<Map<String, Object>> health() {
        Map<String, Object> health = new HashMap<>();
        health.put("status", "UP");
        health.put("timestamp", LocalDateTime.now());
        health.put("service", "config-server");
        health.put("port", 8888);
        return ResponseEntity.ok(health);
    }
}
